r2-core/src/main/java/com/linkedin/r2/transport/http/client/AsyncPoolImpl.java

```diff
@@ -816,6 +831,40 @@ public long getTime()
     }
   }
 
+  private class ObjectCreationTimeoutCallback implements Callback<T>
```

Comment:
This looks like a duplicate of TimeoutCallback. Can you just reuse that?

Reply:
I wanted to throw a retry exception since it is connection related. I have refactored TimeoutCallback to accommodate this need.